web/src/utils/date.ts

```diff
@@ -48,3 +48,50 @@ export const formatDateInUTC00Suffix = (dtISO: string): string => {
   const isoNoon = toISO(DateTime.fromJSDate(dtJS).setZone('UTC'))
   return isoNoon.substring(0, isoNoon.length - 1) + '+00:00'
 }
+
+export const getPrepWeeklyDateRange = (
+  dateOfInterest: string
+): { start: DateTime; end: DateTime } => {
+  const day = DateTime.fromISO(dateOfInterest).weekdayLong
+  let dayOffset = 0
+  switch (day) {
+    case 'Tuesday':
```

Comment:
This will fail of your computers locale happens to be set to a different language.
Rather use the javascript .getDay() function available on the Date object. That will give you the day of the week.
e.g. for Wednesday, getDay() will return 3. So for day.getDay() => 2 day.getDay() <=3, a.getDate() - 1 etc. etc.